app/controllers/spree/omniauth_callbacks_controller.rb

```diff
@@ -34,7 +34,7 @@ def omniauth_callback
       flash[:notice] = I18n.t('devise.sessions.signed_in')
       redirect_back_or_default(account_url)
     else
-      user = Spree::User.find_by_email(auth_hash['info']['email']) || Spree::User.new
+      user = Spree.user_class.find_by_email(auth_hash['info']['email']) || Spree.user_class.new
```

Comment:
@jarednorman I am having second thoughts about the first change on this line too. Is it safe to assume that the class defined by `Spree.user_class` inherits from `Spree::User`? If so, I am going to revert to:
``` ruby
Spree::User.find_by_email(auth_hash['info']['email']) || Spree.user_class.new
```